alg_hmac/alg.go

```diff
@@ -0,0 +1,71 @@
+package alg_hmac
+
+import (
+	"context"
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/hex"
+	"errors"
+
+	"github.com/common-fate/httpsig"
+	"github.com/common-fate/httpsig/contentdigest"
+	"github.com/common-fate/httpsig/signer"
+	"github.com/common-fate/httpsig/verifier"
+)
+
+const HMAC_SHA256 = "hmac-sha256" // Algo is the HMAC algorithm name
+
+// HMAC is a signer and verifier for HMAC digests. It uses crypto/hmac with sha256,
+// and implements the httpsig.Attributer interface
+type HMAC struct {
+	Key   []byte
+	Attrs any
+}
+
+// NewHMAC creates a new HMAC with the provided key
+func NewHMAC(key []byte) *HMAC {
+	return NewHMACWithAttributes(key, nil)
+}
+
+// NewHMACWithAttributes creates a new HMAC with the provided key and attributes
+func NewHMACWithAttributes(key []byte, attrs any) *HMAC {
+	return &HMAC{Key: key, Attrs: attrs}
+}
+
+var _ signer.Algorithm = &HMAC{}
+var _ verifier.Algorithm = &HMAC{}
+var _ httpsig.Attributer = &HMAC{}
+
+func (h *HMAC) Type() string {
+	return HMAC_SHA256
+}
+
+func (h *HMAC) Attributes() any {
+	return h.Attrs
+}
+
+func (h *HMAC) Sign(ctx context.Context, base string) ([]byte, error) {
+	if h.Key == nil {
+		return nil, errors.New("no key provided")
+	}
+	dataHmac := hmac.New(sha256.New, h.Key).Sum([]byte(base))
+	resp := make([]byte, hex.EncodedLen(len(dataHmac)))
+	hex.Encode(resp, dataHmac)
```

Comment:
Should this be hex-encoded? I am thinking we might want to simply return `dataHmac`. The [spec](https://www.rfc-editor.org/rfc/rfc9421.html#section-3.3.3) doesn't mention hex encoding AFAIK. The resulting `[]byte` value is already encoded as a [Byte Sequence](https://www.rfc-editor.org/rfc/rfc8941#section-4.1.8).

Of course, this won't impact anything if both the signer and verifier both use this library, I'm just thinking of use cases where a different verifier implementation is used (say, in another language)

Reply:
Good call, this is an artifact from debugging, I'll drop it